### PR TITLE
samples: drivers: jesd216: check return value of flash_sfdp_read()

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Check the return value of flash_sfdp_read() and print error message if not successful.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

### CODE REVIEW

**Review by pabigot — CHANGES_REQUESTED** [review]
Good idea, but please update the message to something like:

    Read SFDP not supported: device not JESD216-compliant

I'm assuming here that the failure is lack of support for that command, since the flash must have been recognized and the `RDID` command successfully executed to retrieve the JEDEC ID if the device pointer isn't null.

This should be pretty rare, since JESD216 was introduced in 2011.

**Review by pabigot — APPROVED** [review]
Style check might want a blank line before the statement but this works for me.  Thanks.

**Comment by henrikbrixandersen** [comment]
> Good idea, but please update the message to something like:
> 
> ```
> Read SFDP not supported: device not JESD216-compliant
> ```

Thanks, updated.
 
> I'm assuming here that the failure is lack of support for that command, since the flash must have been recognized and the `RDID` command successfully executed to retrieve the JEDEC ID if the device pointer isn't null.
> 
> This should be pretty rare, since JESD216 was introduced in 2011.

Yes. Either that or (perhaps more likely) the (Q)SPI controller hardware does not support the read SFDP command. I came across this when testing the Xilinx AXI QSPI driver (see #27627) with different hardware configurations.

The Xilinx AXI QSPI IP block is configured for a specific type of SPI NOR flash at synthesis time (e.g. Spansion, Winbond, Micron, ...) to reduce the logic and only knows how to handle commands supported by that specific type of memory. Some of these configurations do not support the SFDP read command, (which the driver detects and returns an error).

**Review by de-nordic — APPROVED** [review]

File: samples/drivers/jesd216/src/main.c [hunk 1]
```diff
@@ -241,8 +241,13 @@ void main(void)
...
 	const struct jesd216_sfdp_header *hp = &u.sfdp;
 	int rc = flash_sfdp_read(dev, 0, u.raw, sizeof(u.raw));
-	uint32_t magic = jesd216_sfdp_magic(hp);
+	if (rc != 0) {
+		printf("Read SFDP not supported: device not JESD216-compliant "
+		       "(err %d)\n", rc);
+		return;
+	}
 
+	uint32_t magic = jesd216_sfdp_magic(hp);
```
Comment (ioannisg) [inline samples/drivers/jesd216/src/main.c:250]: yeap, you need a blank line, here as far as I remember

Comment (henrikbrixandersen) [reply samples/drivers/jesd216/src/main.c:251]: @ioannisg Added.

**Review by ioannisg — APPROVED** [review]

**Pull request merged**